### PR TITLE
tails, inits, initTails example implementation

### LINKED ISSUE #213: Add functions inits, tails, and initTails
Many of the supported types have the functions `inits :: a -> [a]` and `tails :: a -> [a]` which are very useful, but they are missing from mono-traversable.

If there is not a native `inits` and `tails` defined for the type there is an easy default implementation leveraging `initMay` and `tailMay` which are already defined in mono-traversable.

I also think including `initTails :: a -> [(a,a)]` (a zip of inits and tails) would be helpful so that types like `[a]` and `Seq a` can have a faster implementation to get the list of splits.

Let me know if this is something you would be interested in me implementing for mono-traversable.

### PR DESCRIPTION
**Pull request by BebeSparkelSparkel**
Here is my example implementation. I have only added:
- `tails`, `inits`, `initTails` to `class IsSequence`
- their default implementations
- added non-default implementations to the List instance

Can you review for correctness and then I will add it to the other instances?

I am unsure how to add this to tests. How should I add the tests?

closes #213

### CODE REVIEW

**Review by snoyberg — CHANGES_REQUESTED** [review]
Some test cases for this would be a nice addition if you don't mind. The code looks correct overall.

**Comment by BebeSparkelSparkel** [comment]
I'll try to add some tests this week

**Review by snoyberg — APPROVED** [review]

**Comment by BebeSparkelSparkel** [comment]
Added tests and fixed error for list instance.

Question: Should `initTails [] == [([],[])]` or `initTails == []`?

**Comment by snoyberg** [comment]
I honestly have no idea, I've never used that function. Is there any prior art you're aware of for a definition?

**Comment by BebeSparkelSparkel** [comment]
I have received comment from @tomjaguarpaw above and [Atijohn](https://www.reddit.com/r/haskell/comments/1b0ke0d/empty_case_for_new_function_inittails/).

Both recommend `initTails [] == [([], [])]`, as now implemented, with good reasoning.

So, this is ready to go.

File: mono-traversable/src/Data/Sequences.hs [hunk 1]
```diff
@@ -472,6 +472,31 @@ class (Monoid seq, MonoTraversable seq, SemiSequence seq, MonoPointed seq) => Is
     splitWhen :: (Element seq -> Bool) -> seq -> [seq]
     splitWhen = defaultSplitWhen
 
+    -- | Returns all the final segments of 'seq' with the longest first.
+    --
+    -- @since ????
```
Comment (snoyberg) [inline mono-traversable/src/Data/Sequences.hs:477]: Once these are updated, I'll be happy to merge and release. Thanks!

**Review by snoyberg — APPROVED** [review]

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**